src/Constraint/Constraint.php

```diff
@@ -84,6 +106,42 @@ public function matches(ConstraintInterface $provider)
         return $provider->matches($this);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function isSubsetOf(ConstraintInterface $constraint)
+    {
+        if ($constraint instanceof EmptyConstraint) {
+            return true;
+        }
+
+        if ($constraint instanceof MultiConstraint) {
+            if ($constraint->isConjunctive()) {
+                foreach ($constraint->getConstraints() as $c) {
+                    if (!$this->isSubsetOf($c)) {
+                        return false;
+                    }
+                }
+
+                return true;
+            } else {
+                foreach ($constraint->getConstraints() as $c) {
+                    if ($this->isSubsetOf($c)) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        if ($this->operator === self::OP_NE) {
+            return (string) $constraint === (string) $this;
```

Comment:
```suggestion
            return $constraint->operator === self::OP_NE && $constraint->version === $this->version && $;
```

Reply:
@jderusse I'm working on a bunch more changes locally so maybe wait before spending too much time reviewing/suggesting things here :)

Reply:
you suggestion looks like invalid code.

Reply:
thanks